### PR TITLE
feat(xdsl.statistics): add HEC, FEC and CRC

### PR DESCRIPTION
**Pull request by Steffy29**
| Question         | Answer
| ---------------- | ---
| Branch?          | `develop`
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | Fix #UXCT-413
| License          | BSD 3-Clause

## Description

Display HEC, FEC and CRC errors statistics for xDSL/Fiber service

### CODE REVIEW

File: packages/manager/apps/telecom/src/app/telecom/pack/xdsl/access/statistics/statistics.controller.js [hunk 1]
```diff
@@ -538,4 +546,100 @@ export default class XdslStatisticsCtrl {
...
+      })
+      .then((stats) => {
+        this.errors.haveSeries = !!(
+          stats.hec.length &&
+          stats.fec.length &&
+          stats.crc.length
+        );
+
+        this.errors.chart = new this.TucChartjsFactory(
+          angular.copy(this.PACK_XDSL_STATISTICS.chart),
```
Comment (marie-j) [inline packages/manager/apps/telecom/src/app/telecom/pack/xdsl/access/statistics/statistics.controller.js:576]: `Object.assign({}, this.PACK_XDSL_STATISTICS.chart)` not sufficient ?

File: packages/manager/apps/telecom/src/app/telecom/pack/xdsl/access/statistics/statistics.controller.js [hunk 2]
```diff
@@ -538,4 +546,100 @@ export default class XdslStatisticsCtrl {
...
+      .all({
+        hec: this.getLinesStatistics('error:hec', period),
+        fec: this.getLinesStatistics('error:fec', period),
+        crc: this.getLinesStatistics('error:crc', period),
+      })
+      .then((stats) => {
+        this.errors.haveSeries = !!(
+          stats.hec.length &&
+          stats.fec.length &&
+          stats.crc.length
```
Comment (marie-j) [inline packages/manager/apps/telecom/src/app/telecom/pack/xdsl/access/statistics/statistics.controller.js:572]: Maybe we can do something like 
```suggestion
          Object.values(stats).every(stat => stat.length)
```
so we don't have to update everytime a stat is added

File: packages/manager/apps/telecom/src/app/telecom/pack/xdsl/access/statistics/statistics.controller.js (hunk 1 above)
Comment (Steffy29) [reply packages/manager/apps/telecom/src/app/telecom/pack/xdsl/access/statistics/statistics.controller.js:576]: Sorry but I don't have the same with this.

File: packages/manager/apps/telecom/src/app/telecom/pack/xdsl/access/statistics/statistics.controller.js [hunk 3]
```diff
@@ -45,6 +45,9 @@ export default class XdslStatisticsCtrl {
     this.attenuation = {
       period: 'preview',
     };
+    this.errors = {
+      period: 'preview',
```
Comment (JeremyDec) [inline packages/manager/apps/telecom/src/app/telecom/pack/xdsl/access/statistics/statistics.controller.js:49]: Maybe we can have a `preview` set in constant ?

File: packages/manager/apps/telecom/src/app/telecom/pack/xdsl/access/statistics/statistics.controller.js [hunk 4]
```diff
@@ -143,6 +148,9 @@ export default class XdslStatisticsCtrl {
    *          - ping
    *          - traffic:upload,
    *          - traffic:download,
+   *          - error:hec,
+   *          - error:fec,
+   *          - error:crc
```
Comment (JeremyDec) [inline packages/manager/apps/telecom/src/app/telecom/pack/xdsl/access/statistics/statistics.controller.js:153]: Should this documentation update be set to https://github.com/ovh/manager/pull/5280/files#diff-b7e0c1728a80adde3a7f3ae67855eed43a53d7454180f76de4f1e78e52dea5f9R129 ?

File: packages/manager/apps/telecom/src/app/telecom/pack/xdsl/access/statistics/statistics.controller.js [hunk 5]
```diff
@@ -538,4 +546,98 @@ export default class XdslStatisticsCtrl {
...
+  getErrorsStatistics(period) {
+    this.errors.loading = true;
+    return this.$q
+      .all({
+        hec: this.getLinesStatistics('error:hec', period),
+        fec: this.getLinesStatistics('error:fec', period),
+        crc: this.getLinesStatistics('error:crc', period),
+      })
+      .then((stats) => {
+        this.errors.haveSeries = !!Object.values(stats).every(
```
Comment (JeremyDec) [inline packages/manager/apps/telecom/src/app/telecom/pack/xdsl/access/statistics/statistics.controller.js:569]:
```suggestion
        this.errors.haveSeries = Object.values(stats).every(
```

File: packages/manager/apps/telecom/src/app/telecom/pack/xdsl/access/statistics/statistics.controller.js [hunk 6]
```diff
@@ -538,4 +546,98 @@ export default class XdslStatisticsCtrl {
...
+          (stat) => stat.length,
+        );
+
+        this.errors.chart = new this.TucChartjsFactory(
```
Comment (JeremyDec) [inline packages/manager/apps/telecom/src/app/telecom/pack/xdsl/access/statistics/statistics.controller.js:573]: Could the code to generate chart + series be in a specific method to be more readable ?

File: packages/manager/apps/telecom/src/app/telecom/pack/xdsl/access/statistics/statistics.controller.js (hunk 4 above)
Comment (Steffy29) [reply packages/manager/apps/telecom/src/app/telecom/pack/xdsl/access/statistics/statistics.controller.js:153]: ??

Comment (JeremyDec) [reply packages/manager/apps/telecom/src/app/telecom/pack/xdsl/access/statistics/statistics.controller.js:153]: @Steffy29 you're adding the details about `error:hec/fec/crc` for the method `getAccessStatistics`, but just under, you're using these flags for the method `getLinesStatistics`, so the documentation should be updated for the method `getLinesStatistics`, not `getAccessStatistics` no?

File: packages/manager/apps/telecom/src/app/telecom/pack/xdsl/access/statistics/statistics.controller.js (hunk 6 above)
Comment (Steffy29) [reply packages/manager/apps/telecom/src/app/telecom/pack/xdsl/access/statistics/statistics.controller.js:573]: It's already done by `TucChartjsFactory`, in this code we add series and labels to display into the generated chart.

File: packages/manager/apps/telecom/src/app/telecom/pack/xdsl/access/statistics/translations/Messages_fr_FR.json [hunk 7]
```diff
@@ -20,5 +20,11 @@
...
   "xdsl_statistics_ping_ms": "{{value}} ms",
   "xdsl_statistics_ping_s": "{{value}} s",
-  "xdsl_statistics_decibel": "{{value}} dB"
-}
\ No newline at end of file
+  "xdsl_statistics_decibel": "{{value}} dB",
+  "xdsl_statistics_errors_title": "Erreurs",
+  "xdsl_statistics_hec_label": "HEC",
+  "xdsl_statistics_fec_label": "FEC",
+  "xdsl_statistics_crc_label": "CRC",
```
Comment (radireddy) [inline packages/manager/apps/telecom/src/app/telecom/pack/xdsl/access/statistics/translations/Messages_fr_FR.json:27]: it is required to translate these? look like they will remain same in all languages. Please move them to a constant file  and use without any translations.

Comment (Steffy29) [reply packages/manager/apps/telecom/src/app/telecom/pack/xdsl/access/statistics/translations/Messages_fr_FR.json:27]: Sorry, but the label could be changed in an other time with a real label. So if I use a constant, I will need to break code.

**Review by radireddy — APPROVED** [review]

**Review by antleblanc — APPROVED** [review]

**Pull request merged**